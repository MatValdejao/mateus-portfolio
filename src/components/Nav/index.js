import React from "react";

function Nav() {
	return (
        <nav>
            <ul className="flex-row">
                <li className="mx-2">
                    <a data-testid="about" href="#about" className="link">
                        About Me
                    </a>
                </li>
                <li className="mx-2">
                    <a data-testid="portfolio" href="#portfolio" className="link">
                        Portfolio
                    </a>
                </li>
                <li className="mx-2">
                    <a data-testid="contact" href="#contact" className="link">
                        Contact
                    </a>
                </li>
            </ul>
        </nav>
	);
}

export default Nav;
