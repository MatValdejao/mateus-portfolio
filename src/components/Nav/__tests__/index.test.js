import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

describe("Nav component", () => {
	it("renders", () => {
		render(<Nav />);
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<Nav />);
		expect(asFragment()).toMatchSnapshot();
	});
});

it("links are visible", () => {
	render(<Nav />);
	expect(screen.getByTestId("about")).toHaveTextContent("About Me");
	expect(screen.getByTestId("portfolio")).toHaveTextContent("Portfolio");
	expect(screen.getByTestId("contact")).toHaveTextContent("Contact");
});
