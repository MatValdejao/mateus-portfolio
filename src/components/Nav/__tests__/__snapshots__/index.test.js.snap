// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="flex-row"
    >
      <li
        class="mx-2"
      >
        <a
          class="link"
          data-testid="about"
          href="#about"
        >
          About Me
        </a>
      </li>
      <li
        class="mx-2"
      >
        <a
          class="link"
          data-testid="portfolio"
          href="#portfolio"
        >
          Portfolio
        </a>
      </li>
      <li
        class="mx-2"
      >
        <a
          class="link"
          data-testid="contact"
          href="#contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
